lib/grppc.gi

```diff
@@ -418,29 +405,31 @@ InstallOtherMethod( GroupWithGenerators,
     IsMultiplicativeElementWithInverseByPolycyclicCollector] ,
     # override methods for `IsList' or `IsEmpty'.
     10,
-    function( gens, id )
-    local G,fam,typ,pcgs;
-
-    fam:= CollectionsFamily( FamilyObj( id ) );
-    if not IsBound(fam!.defaultFinitelyGeneratedGroupWithOneType) then
-      fam!.defaultFinitelyGeneratedGroupWithOneType:=
-	NewType(fam,IsGroup and IsAttributeStoringRep 
-		    and HasGeneratorsOfMagmaWithInverses
-		    and IsFinitelyGeneratedGroup and HasOne
-		    and HasFamilyPcgs and HasHomePcgs);
-    fi;
-    typ:=fam!.defaultFinitelyGeneratedGroupWithOneType;
+function( empty, id )
+local G,fam,typ,pcgs;
 
-    pcgs:=DefiningPcgs(ElementsFamily(fam));
+  fam:= CollectionsFamily( FamilyObj( id ) );
 
-    G:=rec();
-    ObjectifyWithAttributes(G,typ,GeneratorsOfMagmaWithInverses,[],
-                            One,id,FamilyPcgs,pcgs,HomePcgs,pcgs);
+  # pc groups are always finite and gens is finite.
+  typ:=IsGroup and IsAttributeStoringRep 
+        and HasGeneratorsOfMagmaWithInverses and HasOne
+        and IsFinite and IsFinitelyGeneratedGroup
+        and IsTrivial and HasIsNonTrivial
+        and HasIsEmpty and IsSolvableGroup
+        and HasFamilyPcgs and HasHomePcgs and HasGeneralizedPcgs;
+  typ:=NewType(fam,typ);
 
-    SetGroupOfPcgs (pcgs, G);
-    return G;
-end );
+  pcgs:=DefiningPcgs(ElementsFamily(fam));
 
+  G:= rec();
+  ObjectifyWithAttributes( G, typ,
+                          GeneratorsOfMagmaWithInverses, empty,
+                          One, id,
+                          FamilyPcgs,pcgs,HomePcgs,pcgs,GeneralizedPcgs,pcgs);
+
+  #SetGroupOfPcgs (pcgs, G);
```

Comment:
Why is this commented out? It used to be there. Is it not needed anymore? (Why?) If so, I'd just remove that line.